src/test/regress/sql/ch_bench_join_expr.sql

```diff
@@ -0,0 +1,70 @@
+SET citus.next_shard_id TO 1650000;
+CREATE SCHEMA ch_bench_join_expr;
+SET search_path TO ch_bench_join_expr;
+
+SET citus.enable_repartition_joins TO on;
+
+CREATE TABLE order_line (
+    ol_w_id int NOT NULL,
+    ol_d_id int NOT NULL,
+    ol_o_id int NOT NULL,
+    ol_number int NOT NULL,
+    ol_i_id int NOT NULL,
+    ol_delivery_d timestamp NULL DEFAULT NULL,
+    ol_amount decimal(6,2) NOT NULL,
+    ol_supply_w_id int NOT NULL,
+    ol_quantity decimal(2,0) NOT NULL,
+    ol_dist_info char(24) NOT NULL,
+    PRIMARY KEY (ol_w_id,ol_d_id,ol_o_id,ol_number)
+);
+
+CREATE TABLE stock (
+    s_w_id int NOT NULL,
+    s_i_id int NOT NULL,
+    s_quantity decimal(4,0) NOT NULL,
+    s_ytd decimal(8,2) NOT NULL,
+    s_order_cnt int NOT NULL,
+    s_remote_cnt int NOT NULL,
+    s_data varchar(50) NOT NULL,
+    s_dist_01 char(24) NOT NULL,
+    s_dist_02 char(24) NOT NULL,
+    s_dist_03 char(24) NOT NULL,
+    s_dist_04 char(24) NOT NULL,
+    s_dist_05 char(24) NOT NULL,
+    s_dist_06 char(24) NOT NULL,
+    s_dist_07 char(24) NOT NULL,
+    s_dist_08 char(24) NOT NULL,
+    s_dist_09 char(24) NOT NULL,
+    s_dist_10 char(24) NOT NULL,
+    PRIMARY KEY (s_w_id,s_i_id)
+);
+
+CREATE TABLE supplier (
+    su_suppkey int not null,
+    su_name char(25) not null,
+    su_address varchar(40) not null,
+    su_nationkey int not null,
+    su_phone char(15) not null,
+    su_acctbal numeric(12,2) not null,
+    su_comment char(101) not null,
+    PRIMARY KEY ( su_suppkey )
+);
+
+SELECT create_distributed_table('order_line','ol_w_id');
+SELECT create_distributed_table('stock','s_w_id');
+SELECT create_reference_table('supplier');
+
+INSERT INTO supplier SELECT c, 'abc', 'def', c, 'ghi', c, 'jkl' FROM generate_series(0,10) AS c;
+INSERT INTO stock SELECT c,c,c,c,c,c, 'abc','abc','abc','abc','abc','abc','abc','abc','abc','abc','abc' FROM generate_series(1,3) AS c;
+INSERT INTO stock SELECT c, 5000,c,c,c,c, 'abc','abc','abc','abc','abc','abc','abc','abc','abc','abc','abc' FROM generate_series(1,3) AS c; -- mod(2*5000,10000) == 0
+INSERT INTO order_line SELECT c, c, c, c, c, NULL, c, c, c, 'abc' FROM generate_series(0,10) AS c;
+
+-- MVP for CH-BenCHmark queries 7/8/9 where a join with an expression is done between a distributed table and a reference table.
```

Comment:
Could maybe add queries from https://gist.github.com/marcocitus/02a8068a65850549995e8e385bce0018

Reply:
Makes sense, and touches on our conversation yesterday :)

Would it be enough to add these queries without data? At least we know they are planned and executed, but we won't have assurance they provide the correct result.

Ideally we would have data that makes sense for these queries, which so far has been non-trivial in even the reference setup. But I can start working on that.